File: rs/pullcfgwithoverride/src/fs/baz.rs
use crate::config::CfgSrc;
use arc_swap::ArcSwap;
use once_cell::sync::Lazy;

#[derive(Debug)]
pub struct BazCfgInfo {
    pub w: String,
}

pub static BAZ_CFG_SRC: Lazy<ArcSwap<CfgSrc<BazCfgInfo>>> =
    Lazy::new(|| ArcSwap::from_pointee(CfgSrc::<BazCfgInfo>::nil()));

pub fn baz() {
    println!(
        "bazCfgSrc().w.length(): {}",
        BAZ_CFG_SRC.load().get().w.len()
    );
}

File: src/main/kotlin/pullcfgwithoverride/fs/Baz.kt
package pullcfgwithoverride.fs

import pullcfgwithoverride.config.makeCfgSrc

data class BazCfgInfo(
	val w: String
)

var bazCfgSrc = makeCfgSrc<BazCfgInfo>(null)

fun baz() {
	println(bazCfgSrc().w.length);
}
